### PR TITLE
Add documentations

### PR DESCRIPTION
**Pull request by geyyer**
Summary: Add fbgemm_gpu docs similar to https://pytorch.org/torchrec/

Differential Revision: D36271013

### CODE REVIEW

File: fbgemm_gpu/docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	echo.may add the Sphinx directory to PATH.
+	echo.
+	echo.If you don't have Sphinx installed, grab it from
+	echo.https://www.sphinx-doc.org/
+	exit /b 1
+)
+
+%SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
+goto end
+
```
Comment (malfet) [inline fbgemm_gpu/docs/make.bat:30]: Why do you need to add batch file? I though fbgemm GPU is linux only, isn't it?

**Review by malfet — APPROVED** [review]
LGTM, but please address the comments

File: .github/workflows/fbgemm_docs.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    branches:
+      - main
+jobs:
+  build_docs_job:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+         - os: linux.2xlarge
+           python-version: 3.9
```
[the remark this answers is not in the document]
Comment (geyyer) [reply .github/workflows/fbgemm_docs.yml:14]: done

File: fbgemm_gpu/docs/source/index.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+=======================================
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+:ref:`genindex`
+
+Docs
+====
```
Comment (svekars) [inline fbgemm_gpu/docs/source/index.rst:16]:
```suggestion
====================
```
Needs to be as long as the previous line.

File: fbgemm_gpu/docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+.. doxygennamespace:: fbgemm
```
Comment (svekars) [inline fbgemm_gpu/docs/source/index.rst:18]:
```suggestion
.. toctree::
   :maxdepth: 2
   :caption: Contents

   fbgemm_gpu.fbgemm_gpu.rst

File: fbgemm_gpu/docs/source/index.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+   :members:
```
Comment (svekars) [inline fbgemm_gpu/docs/source/index.rst:19]:
```suggestion
```

File: fbgemm_gpu/docs/source/index.rst (hunk 4 above)
Comment (svekars) [reply fbgemm_gpu/docs/source/index.rst:18]: Also add a file named fbgemm_gpu.fbgemm_gpu.rst and add the following to that file:

```
fbgemm_gpu.fbgemm_gpu
=====================

.. automodule:: fbgemm_gpu.fbgemm_gpu

fbgemm_gpu.fbgemm_gpu.batched_unary_embeddings_ops
--------------------------------------------------

.. automodule:: fbgemm_gpu.fbgemm_gpu.batched_unary_embeddings_ops
   :members:
   :show-inheritance:
```

**Review by svekars — APPROVED** [review]

**Pull request closed without merging**